### DISCUSSION TITLE
In a std venv, deactivate has no effect

### DISCUSSION THREAD

**Discussion by mharig** [xonsh/xonsh#4982] [Q&A]
which deactivate
gives:
ForeignShellFunctionAlias(extra_args=(), funcname='deactivate', shell='bash')

xonsh is 0.13.3
I found one issue that didn't match my problem.
It worked fine some time ago.
Any hints?

**Comment by gforsyth** [comment]
If this is a `virtualenv` `venv`, then you can use the `vox` xontrib

`xontrib load vox` and then you can `vox deactivate` (and also `vox activate`)

**Reply by mharig** [reply]
Hm, vox manages its own environments, so I would have to make all my environments new. Not cool.

**Reply by Virgilio-AI** [reply]
I am having the same exact problem. but the biggest of my problems is that box doesn't create a folder like python venv where I can just source and deactivate from. it instead has all global environmets

**Reply by jnoortheen** [reply]
After you create venv with folders, you can activate/deactivate with vox
